packages/nimble-components/src/table/index.ts

```diff
@@ -336,35 +373,103 @@ export class Table<
             return { ...record };
         });
         this.validateWithData(data);
+        this.updateTableOptions(
+            {
+                data,
+                state: {
+                    expanded: true
+                }
+            },
+            true
+        );
+        const expandedState = this.getInflatedExpandedState() as {
+            [key: string]: boolean
+        };
+        for (const collapsedRowId of Object.keys(this.collapsedState)) {
+            if (this.collapsedState[collapsedRowId]) {
+                expandedState[collapsedRowId] = false;
+            }
+        }
         this.updateTableOptions({
-            data
+            state: {
+                expanded: expandedState
+            }
         });
     }
 
+    private getInflatedExpandedState(): TanStackExpandedState {
+        if (this.table.options.state.expanded === true) {
+            const expandedState = {} as { [key: string]: boolean };
+            for (const row of this.table.getRowModel().rows) {
+                expandedState[row.id] = true;
+            }
+            return expandedState;
+        }
+
+        return this.table.options.state.expanded!;
+    }
+
     private refreshRows(): void {
         const rows = this.table.getRowModel().rows;
-        this.tableData = rows.map(row => {
+        this.tableData = rows.map((row, i) => {
+            const groupColumn = this.getGroupRowColumn(i);
             const rowState: TableRowState<TData> = {
                 record: row.original,
-                id: row.id
+                id: row.id,
+                isGrouped: row.getIsGrouped(),
+                isExpanded: row.getIsExpanded(),
+                groupRowValue: groupColumn
+                    ? row?.getLeafRows()[0]?.getValue(row.groupingColumnId!)
```

Comment:
I thought you were going to be able to call `row.getValue(row.groupingColumnId)` here. Did that not work?

Reply:
That does not work. This was the only way I could figure out how to get it to work.

Reply:
I'd be interested in better understanding what about it doesn't work.

Reply:
I apparently am a liar. That works fine.